src/tests/stats/test_mean.f90

```diff
@@ -1,75 +1,90 @@
 program test_mean
-use stdlib_experimental_error, only: assert
-use stdlib_experimental_kinds, only: sp, dp, int32, int64
-use stdlib_experimental_io, only: loadtxt
-use stdlib_experimental_stats, only: mean
-implicit none
+  use stdlib_experimental_error, only: assert
+  use stdlib_experimental_kinds, only: sp, dp, int32, int64
+  use stdlib_experimental_io, only: loadtxt
+  use stdlib_experimental_stats, only: mean
+  implicit none
 
-real(sp), allocatable :: s(:, :)
-real(dp), allocatable :: d(:, :)
+  real(sp), parameter :: sptol = 1.2e-06_sp
+  real(dp), parameter :: dptol = 2.2e-15_dp
 
-real(dp), allocatable :: d3(:, :, :)
-real(dp), allocatable :: d4(:, :, :, :)
+  real(sp) :: s1(3) = [1.0_sp, 2.0_sp, 3.0_sp]
 
+  real(sp), allocatable :: s(:, :)
+  real(dp), allocatable :: d(:, :)
 
-!sp
-call loadtxt("array3.dat", s)
+  real(dp), allocatable :: d3(:, :, :)
+  real(dp), allocatable :: d4(:, :, :, :)
 
-call assert( mean(s) - sum(s)/real(size(s), sp) == 0.0_sp)
-call assert( sum( abs( mean(s,1) - sum(s,1)/real(size(s,1), sp) )) == 0.0_sp)
-call assert( sum( abs( mean(s,2) - sum(s,2)/real(size(s,2), sp) )) == 0.0_sp)
 
+  !sp
+  call loadtxt("array3.dat", s)
 
-!dp
-call loadtxt("array3.dat", d)
+  call assert( abs(mean(s) - sum(s)/real(size(s), sp)) < sptol)
+  call assert( sum( abs( mean(s,1) - sum(s,1)/real(size(s,1), sp) )) < sptol)
+  call assert( sum( abs( mean(s,2) - sum(s,2)/real(size(s,2), sp) )) < sptol)
 
-call assert( mean(d) - sum(d)/real(size(d), dp) == 0.0_dp)
-call assert( sum( abs( mean(d,1) - sum(d,1)/real(size(d,1), dp) )) == 0.0_dp)
-call assert( sum( abs( mean(d,2) - sum(d,2)/real(size(d,2), dp) )) == 0.0_dp)
+  ! check reduction of rank one array to scalar
+  call assert(abs(mean(s1) - sum(s1) / real(size(s1), sp)) < sptol)
+  call assert(abs(mean(s1, dim=1) - sum(s1, dim=1) / real(size(s1, dim=1), sp))&
+      & < sptol)
 
 
-!int32
-call loadtxt("array3.dat", d)
+  !dp
+  call loadtxt("array3.dat", d)
 
-call assert( mean(int(d, int32)) - sum(real(int(d, int32),dp))/real(size(d), dp) == 0.0_dp)
-call assert( sum(abs( mean(int(d, int32),1) - sum(real(int(d, int32),dp),1)/real(size(d,1), dp) )) == 0.0_dp)
-call assert( sum(abs( mean(int(d, int32),2) - sum(real(int(d, int32),dp),2)/real(size(d,2), dp) )) == 0.0_dp)
+  call assert( abs(mean(d) - sum(d)/real(size(d), dp)) < dptol)
+  call assert( sum( abs( mean(d,1) - sum(d,1)/real(size(d,1), dp) )) < dptol)
+  call assert( sum( abs( mean(d,2) - sum(d,2)/real(size(d,2), dp) )) < dptol)
```

Comment:
I used the `==` on purpose since it is the same formula (e.g., `(sum(d)/real(size(d), sp))`) that are used for the function `mean`. 
However, I am also fine with these expressions.

Reply:
Despite being the same formula, there is no warranty for the exact bitwise equivalence at the end. If the first one is calculated in 80-bit float registers and rounded at the end, while the other does every summation with 64 bit floats, the results would be different.

Reply:
I did not think to that possibility. Thank you for the explanations.